### ISSUE TITLE
kubectl debug: Add ability passing additional parameters into profiles

### ISSUE THREAD

**Issue by ardaguclu** [kubernetes/kubectl#1258]
### What would you like to be added?

According to the `kubectl debug` debugging-profiles [enhancement proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/1441-kubectl-debug#debugging-profiles), there are/will be pre-determined and static profiles which user can choose via `profile` flag. This is totally acceptable to avoid lots of feature multiplications and possibly a lot of flag requests to preserve the simplicity of use of debug command.

However, after the discussion in https://github.com/kubernetes/kubectl/issues/1252, the path `debug` command evolving looks like there will be a need to provide a way to at least support some sort of extensibility in debug profiles via flags(not preferable) or `yaml|json` files provided by user(preferable and probably need a KEP as stated in issue). Because even though, there are huge number of profiles being added, still users will want to change some part(ie. `securityContext`?).

Furthermore, downstream CLIs can not easily embrace `kubectl debug` debugging profiles, because these tools might need different kind of profiles possibly populated by user interaction(exactly the case discussed above and does not matter via flags or files). 

I'm proposing a new `ExtraArgs` field. This field will be passed to `applier` and will provide a way to customize debugging profiles. Very likely, this field will be empty for current debugging profiles but it will open a door to customize profiles in some circumstances.

To sum up, what I'm proposing in this draft PR https://github.com/kubernetes/kubernetes/pull/111457/files

### Why is this needed?

As explained above;
- some(rare?) cases, static debugging profiles might be needed to customize by user
- in the future, users can fully customize some template based profiles

**Comment by eddiezane** [comment]
cc @verb

**Comment by CharlieC3** [comment]
Can this be re-opened? Being able to add a `securityContext` to an ephemeral container is important to me, otherwise I cannot use them. My only alternative is to expand the security context of the pod's regular containers, but that comes with some obvious security risks I'd like to avoid in the long-run.

**Comment by verb** [comment]
> Can this be re-opened? Being able to add a `securityContext` to an ephemeral container is important to me, otherwise I cannot use them. My only alternative is to expand the security context of the pod's regular containers, but that comes with some obvious security risks I'd like to avoid in the long-run.

I believe that @ardaguclu is indeed working on a new KEP for this, so I'll reopen.

/reopen
/remove-lifecycle rotten

Feel free to close if the the user-defined profiles is being tracked elsewhere, though.

**Comment by ardaguclu** [comment]
Yes, @verb is right. I have plans to write a KEP for the custom profiling, although I couldn't have found time yet :)

**Comment by ardaguclu** [comment]
As the comments I dropped in https://github.com/kubernetes/kubectl/issues/1364#issuecomment-1552899075, I'm closing this issue to better manage all requests in the KEP.

/close
